src/rpc_client/mod.rs

```diff
@@ -267,12 +284,38 @@ impl fmt::Display for Url {
     }
 }
 
+#[derive(PartialEq, Eq, Debug)]
+pub enum CommunicationProtocol {
+    Http,
+    Ws,
+}
+
+impl TryFrom<&str> for CommunicationProtocol {
+    type Error = anyhow::Error;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        match s {
+            "http" => Ok(Self::Http),
+            "ws" => Ok(Self::Ws),
+            _ => anyhow::bail!("unsupported protocol"),
+        }
+    }
+}
+
 /// Parses a multi-address into a URL
-fn multiaddress_to_url(multiaddr: &Multiaddr, endpoint: &str) -> Url {
+fn multiaddress_to_url(
+    multiaddr: &Multiaddr,
+    endpoint: &str,
+    protocol: CommunicationProtocol,
+) -> Url {
     // Fold Multiaddress into a Url struct
     let addr = multiaddr.iter().fold(
         Url {
-            protocol: DEFAULT_PROTOCOL.to_owned(),
+            protocol: match protocol {
+                CommunicationProtocol::Http => HTTP_PROTOCOL,
+                CommunicationProtocol::Ws => WS_PROTOCOL,
+            }
```

Comment:
Could we use https://docs.rs/strum/latest/strum/derive.IntoStaticStr.html (or other `strum` macro) to get rid of this? Seems like all the information is already there in the enumeration.

Reply:
Done.